specification/eventhub/resource-manager/Microsoft.EventHub/2017-04-01/EventHub.json

```diff
@@ -663,6 +663,55 @@
           }
         }
       }
+    },
+	"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.EventHub/namespaces/{namespaceName}/disasterRecoveryConfigs/CheckNameAvailability": {
```

Comment:
API signature should be providers/{namespace}/checkNameAvailability. The resource type name is specified in the request body, along with the name to check. This would enable the API to work for all resource types if needed.

Reply:
You already have that API. Cant you use that one?

Reply:
This API is for checking the disasterRecoveryConfigs name availability for the name specified in request body.

Reply:
@ravbhatnagar How strongly do you feel about this suggestion? I see your point about have a single CheckNameAvailability API and use the type property to indicate which resource type are we checking availability for.  But if this API were to exist it should be  providers/checkNameAvailability instead of  providers/{namespace}/checkNameAvailability  
because the current URI gives the implication that the API is only for namespaces.

We can implement this suggestion with the next API version.

Reply:
Arun kindly note @ravbhatnagar's suggestion for the next API version in a an issues on VSTS or whatever is convenient for you